### PR TITLE
[FEATURE]: New register schema, helpertext update on form

### LINKED ISSUE #216: [FEATURE]: New register schema, helpertext update on form
## Summary

Register form schema update for names with less than 3 letters, helperText/defaultText on input fields changed , redirect user from register to dashboard route.

### Motivation

Some users can have a name with less than 3 letters
Register form becomes more user friendly

### PR DESCRIPTION
**Pull request by f-morgado**
relates to #216 

## Screenshots (if visual changes)
image
image
image
image

## Proposed Changes

  - Sign up -> SSO email detected -> Login button should say “Login with SSO”
  - Sign up -> Adjust spacing between Login button , “OR” and change email link 
  - Sign up -> Remove first name/ last name 3 char validation
  - Forms -> Only validate and show validation errors when the field is out of focus
  - Password field -> add help to the info label, just switching to red when validation goes wrong
  - Hide terms and conditions checkbox
  - After successful sign up go straight away to dashboard
  
This pull request closes #216

### CODE REVIEW

File: frontend/pages/index.tsx [hunk 1]
```diff
@@ -12,18 +12,24 @@ import Text from '../components/Primitives/Text';
 import { styled } from '../stitches.config';
 import { DASHBOARD_ROUTE } from '../utils/routes';
 
-const CenteredContainer = styled(Flex, {
+const CenteredContainer = styled('div', {
```
Comment (dsousa12) [inline frontend/pages/index.tsx:15]: If you want to clean this page:
1. create a `auth.styles.tsx` on `styles/pages/` folder;
2. move all the `styled` components to that file (remember to export it)
3. import on this page from `auth.styles.tsx`

It's only a suggestion 😉

File: backend/src/modules/users/dto/create.user.dto.ts [hunk 2]
```diff
@@ -15,13 +15,13 @@ export default class CreateUserDto {
 
   @IsString()
   @IsNotEmpty()
-  @MinLength(3)
+  @MinLength(1)
```
Comment (dsousa12) [inline backend/src/modules/users/dto/create.user.dto.ts:18]: Why not use `@MinLength` with 2 chars?

File: frontend/components/auth/SignUp/RegisterForm.tsx [hunk 3]
```diff
@@ -133,6 +136,7 @@ const RegisterForm: React.FC<RegisterFormProps> = ({
 					type="password"
 					icon="eye"
 					iconPosition="right"
+					helperText={isEmpty(valueHelperText) ? undefined : valueHelperText}
```
[the remark this answers is not in the document]
Comment (dsousa12) [reply frontend/components/auth/SignUp/RegisterForm.tsx:139]: #### Suggestions:
- On the password helper text, maybe align the info icon on the top;
- Set the same size for every info icon on this validations;

File: frontend/components/auth/SignUp/RegisterForm.tsx [hunk 4]
```diff
@@ -38,17 +44,12 @@ interface RegisterFormProps {
...
 }
 
-const RegisterForm: React.FC<RegisterFormProps> = ({
-	setShowSignUp,
-	emailName,
-	setCurrentTab,
-	setEmailName
-}) => {
+const RegisterForm: React.FC<RegisterFormProps> = ({ setShowSignUp, emailName, setEmailName }) => {
+	const valueHelperText = msgHelpertext;
```
Comment (RubenMCCarreira) [inline frontend/components/auth/SignUp/RegisterForm.tsx:48]: why are you assign the message to a new const?

File: frontend/components/auth/SignUp/SignUpOptionsForm.tsx [hunk 5]
```diff
@@ -30,23 +30,20 @@ const SignUpOptionsForm: React.FC<SignUpOptionsFormProps> = ({
...
 	};
 
-	// const [ShowLogIn, setShowLogIn] = useState(false);
 	return (
 		<Container direction="column">
 			<Text size="md">
 				The email
 				<Text size="md" fontWeight="medium">
 					{` ${emailName} `}
 				</Text>
```
Comment (RubenMCCarreira) [inline frontend/components/auth/SignUp/SignUpOptionsForm.tsx:39]: try replace by something like this pls
```suggestion
				<Text size="md" fontWeight="medium" css={{ marginHorizontal: '5px' }}>
					{emailName}
				</Text>
```

**Review by dsousa12 — APPROVED** [review]

**Comment by f-morgado** [comment]
Tyvm for the tips and help.

**Pull request merged**